### PR TITLE
Add font smoothing + text link styles

### LINKED ISSUE #108: Text links need improved styles
From [the mockups](https://www.figma.com/file/76mfUcaK4QDlrXElk8MK3H/Twenty-Twenty-Two?node-id=729%3A4866): 

---

image

---

- Links need more refined text-decoration styles, so that the thin font isn't paired with a [horribly thick underline](https://cloudup.com/czdjAkphnKr). (I think we could probably elevate this issue from "enhancement" to a "bug" because of this bullet). 
- Link underlines should become dotted on hover and in their active state.  
- Focus state should be dotted.
- The mockups indicate that large post titles should _not_ have underlines by default, but that their hover and active states should have a solid underline.

May end up being related to: #99 
Upstream issue: https://github.com/WordPress/gutenberg/issues/27075

### PR DESCRIPTION
**Pull request by kjellr**
Fixes #108 and #161. 

Loads the `style.css` file in the editor, and then adds CSS for font smoothing, and for basic text link styles (including hover, active, and focus). These styles do not impose a `text-decoration: underline` rule, so they won't effect or override elements that don't have an underline by default (like the button block).

Each group of rules is accompanied by a description of why it's implemented with CSS, and points to a Gutenberg issue for more details. 

## Screenshots

_Link hover state:_
 
link

Note: I went with a dashed instead of dotted underline so that it better matches the dashed lines used throughout the bird patterns. 

Before|After
---|---
Screen Shot 2021-11-09 at 10 28 52|Screen Shot 2021-11-09 at 10 28 40

### CODE REVIEW

**Comment by kjellr** [comment]
Note that some of the underline rules can be added to theme.json [via this technique](https://github.com/WordPress/twentytwentytwo/issues/108#issuecomment-960503784), but it seems a little delicate to me, and we'd still require CSS for some other text link styles.

File: functions.php [hunk 1]
```diff
@@ -13,17 +13,17 @@
...
 	 *
 	 * @return void
 	 */
 	function twentytwentytwo_support() {
 
+		// Add support for block styles.
 		add_theme_support( 'wp-block-styles' );
 
+		// Enqueue editor styles.
+		add_editor_style( 'style.css' );
```
Comment (jffng) [inline functions.php]: Is it safer to do this?

```suggestion
		add_editor_style( get_template_directory_uri() . '/style.css' );
```

File: style.css [hunk 2]
```diff
@@ -16,3 +16,34 @@ Tags: one-column, custom-colors, custom-menu, custom-logo, editor-style, feature
...
+ */
+
+body {
+	-moz-osx-font-smoothing: grayscale;
+	-webkit-font-smoothing: antialiased;
+}
+
+/*
+ * Text link styles.
+ * Necessary until the following issue is resolved in Gutenberg: ;
```
Comment (jffng) [inline style.css]:
```suggestion
 * Necessary until the following issue is resolved in Gutenberg:
```

File: style.css [hunk 3]
```diff
@@ -16,3 +16,34 @@ Tags: one-column, custom-colors, custom-menu, custom-logo, editor-style, feature
...
+}
+
+a:hover,
+a:focus {
+	text-decoration-style: dashed;
+}
+
+a:active {
+	text-decoration: none;
+}
```
Comment (jffng) [inline style.css]: Every file should end in a newline. 

```suggestion
}

```

**Review by jffng — APPROVED** [review]
Some small changes, otherwise LGTM.

File: functions.php (hunk 1 above)
Comment (kjellr) [reply functions.php]: Turns out this doesn't work. 😄 I'll remove in a subsequent commit.

**Pull request merged**